src/inputs/TextFieldBase.tsx

```diff
@@ -60,15 +71,49 @@ export function TextFieldBase(props: TextFieldBaseProps) {
     onFocus,
     xss,
     endAdornment,
+    startAdornment,
     inlineLabel,
+    borderless = false,
+    contrast = false,
   } = props;
   const errorMessageId = `${inputProps.id}-error`;
   const labelSuffix = getLabelSuffix(required);
   const ElementType: React.ElementType = multiline ? "textarea" : "input";
-  const tid = useTestIds(props, defaultTestId(label || "textField"));
+  const tid = useTestIds(props, defaultTestId(label));
   const [isFocused, setIsFocused] = useState(false);
+  const { hoverProps, isHovered } = useHover({});
   const { focusWithinProps } = useFocusWithin({ onFocusWithinChange: setIsFocused });
 
+  const fieldHeight = 40;
+  const compactFieldHeight = 32;
+
+  const fieldStyles = {
```

Comment:
Hm, this kinda of de-inlines things vs. the usual tachyons/utility css approach, but ah I guess it's because we re-use the styles between the read-only vs. not-read-only side of things.